### PR TITLE
Rename Unadjusted Laplace Approximation sampler to UGLA

### LINKED ISSUE #203: Rename UnadjustedLaplaceApproximation to UGLA

### PR DESCRIPTION
**Pull request by nabriis**
Closes #203

### CODE REVIEW

File: demos/tutorials/Gibbs.py [hunk 1]
```diff
@@ -231,12 +231,12 @@
...
 # ----------------------------------
 #
-# Using the same approach as ealier we can define a Gibbs sampler
+# Using the same approach as earlier we can define a Gibbs sampler
 # for this new hierarchical model. The only difference is that we
 # now need to use a different sampler for :math:`\mathbf{x}` because
 # the ``Linear_RTO`` sampler only works for Gaussian distributions.
 #
-# In this case we use the UnadjustedLaplaceApproximation sampler
+# In this case we use the UGLA (Unadjusted Laplace Approximation) sampler
```
Comment (amal-ghamdi) [inline demos/tutorials/Gibbs.py:239]:
```suggestion
# In this case we use the UGLA (Unadjusted Gaussian Laplace Approximation) sampler
```

File: cuqi/sampler/_laplace_approximation.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def __init__(self, target, x0=None, maxit=50, tol=1e-4, beta=1e-5, rng=None, **k
 
         # Check that prior is Laplace_diff
         if not isinstance(self.target.prior, cuqi.distribution.Laplace_diff):
-            raise ValueError('Unadjusted Laplace approximation requires Laplace_diff prior')
+            raise ValueError('Unadjusted Laplace approximation (UGLA) requires Laplace_diff prior')
```
Comment (amal-ghamdi) [inline cuqi/sampler/_laplace_approximation.py:73]:
```suggestion
            raise ValueError('Unadjusted Gaussian Laplace approximation (UGLA) requires Laplace_diff prior')
```

**Review by amal-ghamdi — APPROVED** [review]
LGTM

**Review by jakobsj — APPROVED** [review]
Any updates to do in https://github.com/CUQI-DTU/Paper-CUQIpy-1-Core from this?  Please check and fix before merging this, other than that looks good.

**Pull request merged**